File: 专题/状态DP/股票问题/714.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    int maxProfit(vector<int>& prices, int fee) {
        int n = prices.size();
        int empty = 0, exist = -1000000, temp;    // exist 改成这个是防止 exist + prices[i] - fee 溢出
        for (int i = 0; i < n; ++i) {
            temp = empty;
            empty = max(empty, exist + prices[i] - fee);
            exist = max(exist, temp - prices[i]);
        }
        return empty;
    }
};

File: 专题/状态DP/股票问题/714.py
class Solution:
    def maxProfit(self, prices: List[int], fee: int) -> int:
        n = len(prices)
        empty = 0
        exist = -10000000000
        for i in range(n):
            temp = empty
            empty = max(exist + prices[i]-fee, empty)
            exist = max(exist, temp - prices[i])
        return empty